controllers/handler/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -157,14 +149,39 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(_ context.Context,
 		err = r.deleteEnactmentForPolicy(request.NamespacedName.Name)
 		return ctrl.Result{}, err
 	}
+	return r.reconcileEnactmentConditions(instance, &log)
+}
+
+func (r *NodeNetworkConfigurationPolicyReconciler) fetchNodeNetworkConfigurationPolicyInstance(
+	instance *nmstatev1.NodeNetworkConfigurationPolicy,
+	request *ctrl.Request,
+	log *logr.Logger,
+) error {
+	err := r.Client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Policy is not found, removing previous enactment if any")
+			err = r.deleteEnactmentForPolicy(request.NamespacedName.Name)
```

Comment:
I don't like that this function, that according to its name merely "fetches" a NNCP also deletes something else.